File: ValidPalindrome.java
package TwoPointers;

public class ValidPalindrome {
    static void main() {
        System.out.println("Two Pointers and Regex to clean ");

        System.out.println(isPalindrome("A man, a plan, a canal: Panama"));
        System.out.println(isPalindrome("race a car"));
        System.out.println(isPalindrome(" "));

    }

    public static boolean isPalindrome(String s) {
        char[] result = s.toLowerCase().replaceAll("[^a-zA-Z0-9]", "").toCharArray() ;
        int i=0; int j=result.length-1;
        while(i <= j){
            if(result[i++] == result[j--]);
            else return false;
        }
        return true;
    }


}

File: src/Solutions/mix/ValidPalindrome.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
#include <ctype.h>

bool isPalindrome(char* s) {
    int i = 0;
    int j = strlen(s) - 1;

    while (i < j) {
        while (i < j && !isalnum((unsigned char)s[i])) {
            i++;
        }
        while (i < j && !isalnum((unsigned char)s[j])) {
            j--;
        }

        if (tolower((unsigned char)s[i]) != tolower((unsigned char)s[j])) {
            return false;
        }

        i++;
        j--;
    }

    return true;
}

int main() {
    char text1[] = "A man, a plan, a canal: Panama";
    char text2[] = "race a car";


    if (isPalindrome(text1)) {
        printf("\"%s\" -> Palindrom!\n", text1);
    } else {
        printf("\"%s\" -> Not Palindrom!\n", text1);
    }

    if (isPalindrome(text2)) {
        printf("\"%s\" -> Palindrom!\n", text2);
    } else {
        printf("\"%s\" -> Not Palindrom!\n", text2);
    }

    return 0;
}